domain/src/main/java/org/oppia/domain/classify/rules/GenericRuleClassifier.kt

```diff
@@ -75,55 +85,71 @@ internal class GenericRuleClassifier private constructor(
 
     abstract fun matches(answer: InteractionObject, inputs: List<InteractionObject>): Boolean
 
-    internal class NoInputMatcherDelegate<T: Any>(
+    internal class NoInputMatcherDelegate<T : Any>(
       private val matcher: NoInputInputMatcher<T>,
       private val extractObject: (InteractionObject) -> T
-    ): MatcherDelegate() {
+    ) : MatcherDelegate() {
       override fun matches(answer: InteractionObject, inputs: List<InteractionObject>): Boolean {
         check(inputs.isEmpty())
         return matcher.matches(extractObject(answer))
       }
     }
 
-    internal class SingleInputMatcherDelegate<T: Any>(
+    internal class SingleInputMatcherDelegate<T : Any>(
       private val matcher: SingleInputMatcher<T>,
       private val extractObject: (InteractionObject) -> T
-    ): MatcherDelegate() {
+    ) : MatcherDelegate() {
       override fun matches(answer: InteractionObject, inputs: List<InteractionObject>): Boolean {
         check(inputs.size == 1)
         return matcher.matches(extractObject(answer), extractObject(inputs.first()))
       }
     }
 
-    internal class MultiTypeSingleInputMatcherDelegate<AT: Any, IT: Any>(
+    internal class MultiTypeSingleInputMatcherDelegate<AT : Any, IT : Any>(
       private val matcher: MultiTypeSingleInputMatcher<AT, IT>,
       private val extractAnswerObject: (InteractionObject) -> AT,
       private val extractInputObject: (InteractionObject) -> IT
-    ): MatcherDelegate() {
+    ) : MatcherDelegate() {
       override fun matches(answer: InteractionObject, inputs: List<InteractionObject>): Boolean {
         check(inputs.size == 1)
         return matcher.matches(extractAnswerObject(answer), extractInputObject(inputs.first()))
       }
     }
 
-    internal class DoubleInputMatcherDelegate<T: Any>(
+    internal class DoubleInputMatcherDelegate<T : Any>(
       private val matcher: DoubleInputMatcher<T>,
       private val extractObject: (InteractionObject) -> T
-    ): MatcherDelegate() {
+    ) : MatcherDelegate() {
       override fun matches(answer: InteractionObject, inputs: List<InteractionObject>): Boolean {
         check(inputs.size == 2)
         return matcher.matches(extractObject(answer), extractObject(inputs[0]), extractObject(inputs[1]))
       }
     }
+
+    internal class MultiTypeDoubleInputMatcherDelegate<AT : Any, ITF : Any, ITS : Any>(
+      private val matcher: MultiTypeDoubleInputMatcher<AT, ITF, ITS>,
+      private val extractAnswerObject: (InteractionObject) -> AT,
+      private val extractFirstParamObject: (InteractionObject) -> ITF,
+      private val extractSecondParamObject: (InteractionObject) -> ITS
+
```

Comment:
Nit: remove empty newline for consistency.

Feel free to mark this as resolved once you fix it.

Reply:
done